--- dbug/src/scripts/explorer.rb	2013-01-23 20:11:44.478865370 -0500
+++ explorer.rb	2013-01-23 20:10:20.222868385 -0500
@@ -296,6 +296,7 @@
   options[:strategy] = "0"
   options[:arbiter] = nil
   options[:ilayer] = nil
+  options[:xtern] = "#{ENV[ 'XTERN_ROOT' ]}/dync_hook/interpose.so"
   options[:logs] = "#{Dir.pwd}/dbug-logs"
   opts.on( '-p', '--prefix PATH',
   'Provide prefix of dBug installation (default: /usr)' ) do |p|
@@ -433,7 +434,8 @@
   FileUtils.move("strategy", "#{$logs}/strategy-#{iter}")
 
   # Start the test
-  ENV[ 'LD_PRELOAD' ] = $ilayer
+  ENV[ 'LD_PRELOAD' ] = "#{options[:xtern]}"
+  puts "checking LD_PRELOAD = #{ENV[ 'LD_PRELOAD' ]}"
   system(*(ARGV[0]))
   ENV[ 'LD_PRELOAD' ] = nil
 
